### DISCUSSION TITLE
Does Renovate send my code to the online services or keeps it local?

### DISCUSSION THREAD

**Discussion by BrunoVT1992** [renovatebot/renovate#23290] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

npx renovate

### If you're self-hosting Renovate, select which platform you are using.

Azure DevOps (dev.azure.com)

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Wanted end result.

We are looking into using the Renovate npx package. But first we are doing a small security audit. Our main concern is that Renovate sends our code to an outside service to do the analysis. We cant really find a clear answer in the docs or forums about how this behaviour works. Can someone please explain if Renovate does send code to its own services and if so how much off the code?

### What you tried so far.

Reading the docs, forums and asking chat gpt.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment, marked as answer]
Renovate doesn't care about code, it cares about dependencies. It doesn't really send anything anywhere other than what it needs to do to look up versions and release notes. Eg it must "send" the package name to the npm registry if it wants to get back the list of versions.
